scripts/install/calibre-web.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+# Flying Sausages 2020 for swizzin
+#Calibre-web installer
+
+calibrewebdir="/opt/calibre-web"
+calibrewebuser="calibreweb" # or make this master user?
+
+log='/root/logs/swizzin.log'
+
+function _install_dependencies_calibreweb () {
+    echo "Installing system dependencies"
+    apt-get install unzip python3 python3-pip imagemagick -y >> $log 2>&1
+}
+
+function _install_calibreweb () {
+    echo "Downloading source code archive" | tee -a $log
+    dlurl=$(curl -s https://api.github.com/repos/janeczku/calibre-web/releases/latest | grep "browser_download_url" | head -1 | cut -d\" -f 4)
+    # shellcheck disable=SC2181
+    if [[ $? != 0 ]]; then
+        echo "Failed to query github" | tee -a $log
+        exit 1
+    fi
+
+    # mkdir -p "$calibrewebdir"
+    wget "${dlurl}" -O /tmp/calibreweb.zip >> $log 2>&1
+    # shellcheck disable=SC2181
+    if [[ $? != 0 ]]; then
+        echo "Failed to download binary" | tee -a $log
+        exit 1
+    fi
+
+    echo "Extracting archive" | tee -a $log
+    unzip /tmp/calibreweb.zip -d /tmp/calibrewebdir >> $log 2>&1
+    subdir=$(ls /tmp/calibrewebdir)
+    mv /tmp/calibrewebdir/"$subdir" $calibrewebdir
+
+    echo "Creating users and setting permissions" | tee -a $log
+    useradd $calibrewebuser --system -d "$calibrewebdir" >> $log 2>&1
+    
+    if [[ -z $calibredbuser ]]; then 
+        calibredbuser=$(cut -d: -f1 < /root/.master.info)
+    fi
+    usermod -a -G "${calibredbuser}" $calibrewebuser >> $log 2>&1
+
+    sudo chown -R $calibrewebuser:$calibrewebuser $calibrewebdir
+
+    echo "Installing python dependencies" | tee -a $log
+    runuser -l $calibrewebuser -c "python3 -m pip install --target vendor -r $calibrewebdir/requirements.txt --upgrade" >> $log 2>&1
+    #fuck ldap. all my homies hate ldap
+    sed '/ldap/Id' -i $calibrewebdir/optional-requirements.txt
```

Comment:
Yeah, for the best. You want LDAP? They you know enough to know what you are doing and can do it yourself. Leave that stank out of this script.